src/utilities/plugin-utils.ts

```diff
@@ -110,10 +108,14 @@ export const validateEmbeddablePluginContextForWrappedEmbeddable =
 };
 
 // loads the learner plugin state into the firebase write-through cache
-// export const loadLearnerPluginState = async (activities: Activity[], teacherEditionMode: boolean) => {
-//   const plugins = findUsedPlugins(activities, teacherEditionMode);
-//   return await Promise.all(plugins.map(async (plugin) => await getLearnerPluginState(plugin.id)));
-// };
+export const loadLearnerPluginState = async (activities: Activity[], teacherEditionMode: boolean) => {
+  const approvedScripts = findUsedApprovedScripts(activities);
+  // PJ 09/19/2021: This doesn't seem to make sense. Currently, the state is saved and restored per approved script.
+  // It should be saved and restored per plugin instance. It works with Glossary only because there's one glossary
+  // instance per activity. Fixing this would destroy students data, so I'm not doing this. But it should be handled
+  // if we ever add more plugins that save their state.
+  return await Promise.all(approvedScripts.map(async (approvedScriptInfo) => await getLearnerPluginState(approvedScriptInfo.id)));
+};
```

Comment:
I'm surprised this is in this PR. It was added to master in a different PR.

Reply:
It got added to new-sections branch in the last rebase to master.